src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml

```diff
@@ -51,16 +38,14 @@
         <service id="messenger.transport.default_decoder" alias="messenger.transport.serialize_message_with_type_in_headers" public="true" />
 
         <!-- Logging & Debug -->
-        <service id="messenger.middleware.debug.logging" class="Symfony\Component\Messenger\Debug\LoggingMiddleware">
+        <service id="messenger.middleware.logging" class="Symfony\Component\Messenger\Middleware\LoggingMiddleware">
```

Comment:
IMHO, each middleware should be abstract and declined for each bus in order to be able to decorate them independently if needed.

Reply:
What's the use case of decorating a middleware?

Reply:
There could be many. The one I may suggest after this one is an `ActivationMiddlewareDecorator`, that will decorate a middleware to conditionally enable it. For instance, in debug mode only. That could solve https://github.com/symfony/symfony/issues/26901 differently.